src/llm/io_processing/output_parser.hpp

```diff
@@ -87,7 +87,7 @@ class OutputParser {
     std::optional<rapidjson::Document> parseChunk(const std::string& chunkResponse, const bool toolsAvailable, ov::genai::GenerationFinishReason finishReason);
 
     bool requiresStreamingWithSpecialTokens() const {
-        return (reasoningParser && reasoningParser->requiresStreamingWithSpecialTokens()) &&
+        return (reasoningParser && reasoningParser->requiresStreamingWithSpecialTokens()) ||
                (toolParser && toolParser->requiresStreamingWithSpecialTokens());
```

Comment:
@mzegla when i implemented it i remember your comment why it should really be `&&` instead of `||`, do you remember what was the reason?

Reply:
This has been implemented this way to make sure we don't allow two parsers with different special tokens approach as they will receive the same model output, so they must both either require it or not. 
I guess for this case, where we don't have reasoning parser but want to require special tokens for tool parser we should modify this function like:
```
if (!reasoningParser) {
	return toolParser && toolParser->requiresStreamingWithSpecialTokens();
} else if (!toolParser) {
	return reasoningParser && reasoningParser->requiresStreamingWithSpecialTokens();
} else {
	return (reasoningParser && reasoningParser->requiresStreamingWithSpecialTokens()) &&
    	(toolParser && toolParser->requiresStreamingWithSpecialTokens());
}
```